### PR TITLE
Fix cblas_nrm2 for complex types

### LINKED ISSUE #389: internal implementation of cblas_nrm2 is broken for complex types
```
> x = NMatrix.new([4,1], [2,-4,3,5], dtype: :float64)
> NMatrix::BLAS.nrm2(x, 1, 3)
=> 5.385164807134504
> x = NMatrix.new([4,1], [2,-4,3,5], dtype: :complex128)
> NMatrix::BLAS.nrm2(x, 1, 3)
=> 4.0
```

https://github.com/SciRuby/nmatrix/blob/master/ext/nmatrix/math/nrm2.h

The ATLAS version works fine.

### PR DESCRIPTION
**Pull request by lokeshh**
Pass correct parameters to nrm2_complex_helper

Fixes #389

### CODE REVIEW

**Comment by v0dro** [comment]
Could you write an accompanying test that proves that the hand-coded version returns the same answer as the BLAS version? Preferably use the example raised in the issue.

Also rebase your commits later.

**Comment by lokeshh** [comment]
I do not know what the function should do in case there's a complex number in the vector. For example if the vector is `[1, 1+2i, 3]`, what's the output should be?

**Comment by mohawkjohn** [comment]
@lokeshh It is defined here, in the comments: http://for2html.sourceforge.net/BLAS/dznrm2.html

```
#  DZNRM2 returns the euclidean norm of a vector via the function
#  name, so that
#
#     DZNRM2 := sqrt( conjg( x' )*x )
```

File: spec/blas_spec.rb [hunk 1]
```diff
@@ -186,9 +186,15 @@
...
 
-        x = NMatrix.new([4,1], [2,-4,3,5], dtype: dtype)
+        if [:complex64, :complex128].include? dtype
+          x = NMatrix.new([3,1], [Complex(1,2),Complex(3,4),Complex(0,6)], dtype: dtype)
+          nrm2 = (1**2 + 2**2 + 3**2 + 4**2 + 6**2)**0.5
+        else
+          x = NMatrix.new([4,1], [2,-4,3,5], dtype: dtype)
+          nrm2 = 5.385164807134504
+        end
+
```
Comment (lokeshh) [inline spec/blas_spec.rb]: Do these tests look better?

File: spec/blas_spec.rb [hunk 2]
```diff
@@ -186,9 +186,15 @@
...
 
-        x = NMatrix.new([4,1], [2,-4,3,5], dtype: dtype)
+        if dtype =~ /complex/
+          x = NMatrix.new([3,1], [Complex(1,2),Complex(3,4),Complex(0,6)], dtype: dtype)
+          nrm2 = 8.12403840463596
+        else
+          x = NMatrix.new([4,1], [2,-4,3,5], dtype: dtype)
+          nrm2 = 5.385164807134504
+        end
+
```
Comment (lokeshh) [inline spec/blas_spec.rb]: Does this look good?

**Pull request merged**